CRM/Event/Form/ParticipantView.php

```diff
@@ -100,6 +100,16 @@ public function preProcess() {
     }
     $participantStatuses = CRM_Event_PseudoConstant::participantStatus();
 
+    // CRM-20879: Show 'Transfer or Cancel' option beside 'Change fee selection'
+    //  only if logged in user have 'edit event participants' permission and
+    //  participant status is not Cancelled or Transferred
+    if (CRM_Core_Permission::check('edit event participants') && !in_array($status, array('Cancelled', 'Transferred'))) {
+      $this->assign('transferOrCancelLink', CRM_Utils_System::url(
+        'civicrm/event/selfsvcupdate',
+        sprintf("reset=1&pid=%d&is_backoffice=1&cs=%s", $participantID, CRM_Contact_BAO_Contact_Utils::generateChecksum($contactID, NULL, 'inf'))
```

Comment:
It might be clearer to pass the $query as an array rather than sprintf (which I can't quite figure the purpose of. In general I think the array format is cleaner in 80% of cases.